models/mart/fct_daily_electricity_usage_carbon_emission.sql

```diff
@@ -18,10 +18,36 @@ weighted_daily_intensity AS (
 SELECT
   ahi.INTENSITY_DATE,
   ahi.REGION_NAME,
+  huf.PREMISE_TYPE,
   SUM(ahi.HOURLY_KGCO2*huf.FACTOR) AS "WEIGHTED_DAILY_KGCO2"
 FROM average_hourly_intensity ahi
 LEFT JOIN {{ ref('stg_additional_modelling_resources__hourly_usage_factors') }} huf ON huf.HOUR=ahi.INTENSITY_HOUR
-GROUP BY 1,2
+AND (
+  CASE
+    WHEN dayname(ahi.INTENSITY_DATE) IN ('Mon', 'Tue', 'Wed', 'Thu', 'Fri') THEN TRUE
+    WHEN dayname(ahi.INTENSITY_DATE) IN ('Sat', 'Sun') THEN FALSE
+  ELSE NULL END
+) = huf.WEEKDAY
+GROUP BY 1,2,3
+),
+factored_locations as (
+SELECT DISTINCT PREMISE_TYPE
+FROM {{ ref('stg_additional_modelling_resources__hourly_usage_factors') }}
+),
+company_locations as (
+  SELECT
+  cd.LOCATION_SIZE_SQ_FT,
+  cd.LOCATION_ID,
+  cd.LOCATION,
+  cd.LOCATION_LATITUDE,
+  cd.LOCATION_LONGITUDE,
+  cd.ABBREVIATED_POST_CODE,
+  CASE
+    WHEN fl.PREMISE_TYPE IS NULL THEN 'Office'
+    ELSE fl.PREMISE_TYPE
+  END AS PREMISE_TYPE_FOR_JOINING --when the client has location types for which we do not have factors, we just use the Office factors
+  FROM {{ ref('stg_operational_data__company_locations') }} CD
+  LEFT JOIN factored_locations FL on FL.PREMISE_TYPE=cd.LOCATION
 )
 SELECT
   ac.INTENSITY_DATE as DATE,
```

Comment:
perhaps add the estimated daily `kwh_usage` to the final select? i know it wasn't there before, but now I think it would be useful